lib/llvm-backend/src/code.rs

```diff
@@ -1071,18 +1300,262 @@ impl FunctionCodeGenerator<CodegenError> for LLVMFunctionCodeGenerator {
+                    .into_vector_value();
+                let res = builder.build_int_add(v1, v2, &state.var_name());
+                let res = builder.build_bitcast(res, intrinsics.i128_ty, "");
+                state.push1(res);
+            }
+            Operator::I16x8Add => {
+                let (v1, v2) = state.pop2()?;
+                let v1 = builder
+                    .build_bitcast(v1, intrinsics.i16x8_ty, "")
+                    .into_vector_value();
+                let v2 = builder
+                    .build_bitcast(v2, intrinsics.i16x8_ty, "")
+                    .into_vector_value();
+                let res = builder.build_int_add(v1, v2, &state.var_name());
+                let res = builder.build_bitcast(res, intrinsics.i128_ty, "");
+                state.push1(res);
+            }
+            Operator::I32x4Add => {
+                let (v1, v2) = state.pop2()?;
+                let v1 = builder
+                    .build_bitcast(v1, intrinsics.i32x4_ty, "")
+                    .into_vector_value();
+                let v2 = builder
+                    .build_bitcast(v2, intrinsics.i32x4_ty, "")
+                    .into_vector_value();
+                let res = builder.build_int_add(v1, v2, &state.var_name());
+                let res = builder.build_bitcast(res, intrinsics.i128_ty, "");
+                state.push1(res);
+            }
+            Operator::I64x2Add => {
+                let (v1, v2) = state.pop2()?;
+                let (v1, v2) = (v1.into_int_value(), v2.into_int_value());
+                let v1 = builder
+                    .build_bitcast(v1, intrinsics.i64x2_ty, "")
+                    .into_vector_value();
+                let v2 = builder
+                    .build_bitcast(v2, intrinsics.i64x2_ty, "")
+                    .into_vector_value();
+                let res = builder.build_int_add(v1, v2, &state.var_name());
+                let res = builder.build_bitcast(res, intrinsics.i128_ty, "");
+                state.push1(res);
+            }
+            Operator::I8x16AddSaturateS => {
+                let (v1, v2) = state.pop2()?;
+                let (v1, v2) = (v1.into_int_value(), v2.into_int_value());
+                let v1 = builder.build_bitcast(v1, intrinsics.i8x16_ty, "");
+                let v2 = builder.build_bitcast(v2, intrinsics.i8x16_ty, "");
+                let res = builder
+                    .build_call(intrinsics.sadd_sat_i8x16, &[v1, v2], "")
+                    .try_as_basic_value()
+                    .left()
+                    .unwrap();
+                let res = builder.build_bitcast(res, intrinsics.i128_ty, "");
+                state.push1(res);
+            }
+            Operator::I16x8AddSaturateS => {
+                let (v1, v2) = state.pop2()?;
+                let (v1, v2) = (v1.into_int_value(), v2.into_int_value());
+                let v1 = builder.build_bitcast(v1, intrinsics.i16x8_ty, "");
```

Comment:
Should we name all the `""` name parameters all over? I thought they may be outputted to ll?

Reply:
Depends on LLVM. A debug LLVM preserves instruction names and a release LLVM discards them (the symbol table for instruction names can be rather slow!). Each opcode implementation should copy the `state.var_name()` to one of the LLVM instructions, and that's usually the same point where we go from naming the rust variable an input value to a result. I think I put names on any instruction that wasn't obvious.

Reply:
@bjfish I noticed that there were some opcodes which weren't using the `state.var_name()` at all. That should be fixed, please point it out if it isn't!